frappe/desk/doctype/notification_log/notification_log.py

```diff
@@ -88,24 +88,33 @@ def enqueue_create_notification(users: list[str] | str, doc: dict):
 		users = [user.strip() for user in users.split(",") if user.strip()]
 	users = list(set(users))
 
-	frappe.enqueue(
-		"frappe.desk.doctype.notification_log.notification_log.make_notification_logs",
-		doc=doc,
-		users=users,
-		now=frappe.flags.in_test,
-	)
+	if frappe.flags.in_test:
+		make_notification_log(doc, users)
+	else:
+		frappe.local.system_notifications_queue.append((doc, users))
+		if flush_system_notifications not in frappe.db.after_commit._functions:
```

Comment:
```suggestion
		if not frappe.local.system_notifications_queue:
```

Can do this instead of relying on internal attribute. Line will have to move up before `append`.

Reply:
This pattern is very common though, maybe we should have `deduplicate` or `once` flag on callback manager :smile: